pkg/apis/unikorn/v1alpha1/types.go

```diff
@@ -272,6 +335,30 @@ type KubernetesClusterApplicationBundle struct {
 	Status            ApplicationBundleStatus `json:"status,omitempty"`
 }
 
+// VirtulKubernetesClusterApplicationBundleList defines a list of application bundles.
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type VirtualKubernetesClusterApplicationBundleList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []VirtualKubernetesClusterApplicationBundle `json:"items"`
+}
+
+// VirtualKubernetesClusterApplicationBundle defines a bundle of applications related with a particular
+// custom resource.
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:scope=Namespaced,categories=unikorn
+// +kubebuilder:printcolumn:name="version",type="string",JSONPath=".spec.version"
+// +kubebuilder:printcolumn:name="preview",type="string",JSONPath=".spec.preview"
+// +kubebuilder:printcolumn:name="end of life",type="string",JSONPath=".spec.endOfLife"
+// +kubebuilder:printcolumn:name="age",type="date",JSONPath=".metadata.creationTimestamp"
+type VirtualKubernetesClusterApplicationBundle struct {
```

Comment:
How come there another type for this, rather than just having one `ApplicationBundle`; is it so that the bundles that can be applied in different places are disjoint sets?

Reply:
It did used to be unified, but then you have a few problems

* You can use a bundle for say a kubernetes cluster as a virtual cluster and it will break
* It's more confusing when you `kubectl get` and have to grok what beings to what

Type safety in essence.  Eventually, we may want to hard code the required applications in the CRD rather than just having an arbitrary list, then even less can go wrong.

Reply:
> Type safety in essence

OK, I can see why you'd want to keep bundles intended for virtual clusters separate.